src/EditorFeatures/CSharpTest/EncapsulateField/EncapsulateFieldTestState.cs

```diff
@@ -58,7 +59,9 @@ public static EncapsulateFieldTestState Create(string markup)
         public void Encapsulate()
         {
             var args = new EncapsulateFieldCommandArgs(_testDocument.GetTextView(), _testDocument.GetTextBuffer());
-            var commandHandler = new EncapsulateFieldCommandHandler(Workspace.GetService<ITextBufferUndoManagerProvider>(),
+            var commandHandler = new EncapsulateFieldCommandHandler(
+                Workspace.ExportProvider.GetExportedValue<IThreadingContext>(),
+                Workspace.GetService<ITextBufferUndoManagerProvider>(),
```

Comment:
Is there a difference between GetService and ExportProvider.GetExportedValue?

Reply:
ExportProvider.GetExportedValue obtains the single value directly from the MEF catalog. GetService gets a workspace service in the context of a specific workspace (accounting for things like ServiceLayer).

Reply:
I got confused as well. so this GetService actually return MEF Service like @jasonmalinowski indicates not Workspace services...

Workspace.Services.GetService returning workspace service and Workspace.GetService returning MEF service is so confusing..

Reply:
there is so many references on GetService so i won't change its naming in this PR...

Reply:
> so this GetService

Ah, yes, this is the `GetService` defined in test code.

Reply:
@heejaechang Sounds good, do we want a follow-up then to just always require Workspace.ExportProvider.GetExportedValue? Although longer I'll agree it's less confusing!